### PR TITLE
Use `syn-teams.libsonnet` for alert routing discovery and alert patching

### PR DESCRIPTION
**Pull request by simu**
This PR switches the alert routing discovery and alert patching to use the new `syn-teams.libsonnet` Jsonnet library. This library is hosted at https://github.com/projectsyn/jsonnet-libs and is intended to be used by all components which interact with `parameters.syn.owner` and `parameters.syn.teams`.

## TODO

- [x] Switch to `version: 'main'` once https://github.com/projectsyn/jsonnet-libs/pull/1 is merged

## Checklist

- [x] The PR has a meaningful title. It will be used to auto-generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: jsonnetfile.jsonnet
Comment (haasad) [inline jsonnetfile.jsonnet:11]: Do you want to wait until this branch is merged?

File: lib/openshift4-monitoring-alert-patching.libsonnet
Comment (haasad) [inline lib/openshift4-monitoring-alert-patching.libsonnet:102]: Can you explain how this variable works? I can't find where it's used

File: jsonnetfile.jsonnet
Comment (simu) [reply jsonnetfile.jsonnet:11]: Yes, see the TODO section in the description.

**Review by haasad — APPROVED** [review]
good to merge when the leftover variable is removed :+1:

**Pull request merged**
